File: easy/move_zeroes/move_zeroes.go
package main

import "fmt"

func moveZeroes(nums []int) {
	numberOfZeroes := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] == 0 {
			numberOfZeroes++
		} else {
			nums[i-numberOfZeroes] = nums[i]
		}
	}

	for numberOfZeroes > 0 {
		nums[len(nums)-numberOfZeroes] = 0
		numberOfZeroes--
	}
	return
}

func main() {
	nums := []int{0, 1, 0, 3, 12}
	moveZeroes(nums)
	fmt.Println("Solution 1 : ", nums)

	nums = []int{0, 0, 1}
	moveZeroes(nums)
	fmt.Println("Solution 2 : ", nums)

	nums = []int{0}
	moveZeroes(nums)
	fmt.Println("Solution 3 : ", nums)
}

File: easy/array/move_zeroes/move_zeroes.py
import copy


class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        number_of_zeroes = 0
        for i in range(0, len(nums)):
            if nums[i] == 0:
                number_of_zeroes += 1
            else:
                nums[i-number_of_zeroes] = nums[i]
        while number_of_zeroes > 0:
            nums[len(nums)-number_of_zeroes] = 0
            number_of_zeroes -= 1
        return


s = Solution()

nums = [0, 1, 0, 3, 12]
s.moveZeroes(nums)
print("Solution 1 : ", nums)

nums = [0, 0, 1]
s.moveZeroes(nums)
print("Solution 2 : ", nums)

nums = [1]
s.moveZeroes(nums)
print("Solution 3 : ", nums)
